engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java

```diff
@@ -185,7 +188,7 @@ public void readInputWithMissedValues() throws Exception {
     TextFileInput input = StepMockUtil.getStep( TextFileInput.class, TextFileInputMeta.class, "test" );
     List<Object[]> output = TransTestingUtil.execute( input, meta, data, 2, false );
     TransTestingUtil.assertResult( new Object[] { "1", "1", "1" }, output.get( 0 ) );
-    TransTestingUtil.assertResult( new Object[] { "2", "1", "2" }, output.get( 1 ) );
+    TransTestingUtil.assertResult( new Object[] { "2", "", "2" }, output.get( 1 ) );
```

Comment:
I had another look into this test and the expected is, in fact, to have { "2", **"1"**, "2" } and not { "2", **""**, "2" }!
If we are, now, getting the latter, we are probably introducing a regression somewhere.
I'm removing my previous approval as this whole development seems to need a better analysis.

Reply:
This is one of the two tests that are failling.

Reply:
I found this comment over the method:
 /*
   * Do not use this method directly! It is for tests!
   */
  @Deprecated
  boolean isNull( Object data, boolean emptyStringDiffersFromNull ) throws KettleValueException {
  
  This method had a wrong comparison with the property also.

Reply:
Code reviewed, and test put back to original state